app/scripts/modules/amazon/src/serverGroup/configure/ServerGroupShardingUtils.ts

```diff
@@ -0,0 +1,110 @@
+import { chain, isEmpty, partition, sortBy, times, toPairs } from 'lodash';
+
+export interface ParsedShards {
+  shards: { [shardIndex: string]: string };
+  other: string;
+}
+
+function createEmpty(dimensions: number, other = ''): ParsedShards {
+  const result: ParsedShards = { shards: {}, other };
+  times(dimensions).forEach(index => (result.shards[`x${index + 1}`] = ''));
+
+  return result;
+}
+
+/**
+ * Given a free form details string, this function parses, validates and returns the shards along with remaining
+ * detail string. This function also accepts a `dimension` param that restricts the number of accepted shard
+ * identifiers.
+ *
+ * freeFormDetails = 'x1Shard1-x2Shard2-x3Shard3-Other', dimensions = 3
+ * Returns: {shards: {x1: 'Shard1', x2: 'Shard2', x3: 'Shard3'}, other: 'Other'}
+ *
+ * freeFormDetails = 'x2Shard2-x1Shard1-x3Shard3-Other', dimensions = 2
+ * Returns: {shards: {x1: 'Shard1', x2: 'Shard2'}, other: 'x3Shard3-Other'}
+ */
+export function parse(freeFormDetails: string, dimensions = 2): ParsedShards {
```

Comment:
A bit of a nit-pick, but naming this `parseShard` or `getParsedShard` would be helpful since this is specific to shard rules and parsing could also happen to JSON objects, other strings, etc.

Reply:
agreed, especially since we tend to re-export all symbols up to the top level.  I'm starting to reconsider this practice because it makes for a very crowded namespace, and potential for collisions.